### ISSUE TITLE
Boot repl does not work with Clojure 1.7.0-alpha4

### ISSUE THREAD

**Issue by grammati** [boot-clj/boot#82]
Here is what happens:

```
% BOOT_CLOJURE_VERSION=1.7.0-alpha4 boot repl
WARNING: cat already refers to: #'clojure.core/cat in namespace: clojure.core.reducers, being replaced by: #'clojure.core.reducers/cat
WARNING: cat already refers to: #'clojure.core/cat in namespace: clojure.core.reducers, being replaced by: #'clojure.core.reducers/cat
nREPL server started on port 57244 on host 0.0.0.0 - nrepl://0.0.0.0:57244
WARNING: cat already refers to: #'clojure.core/cat in namespace: net.cgrand.parsley.fold, being replaced by: #'net.cgrand.parsley.fold/cat
REPL-y 0.3.5, nREPL 0.2.6
Clojure 1.7.0-alpha4
Java HotSpot(TM) 64-Bit Server VM 1.8.0_25-b17
        Exit: Control+D or (exit) or (quit)
    Commands: (user/help)
        Docs: (doc function-name-here)
              (find-doc "part-of-name-here")
Find by Name: (find-name "part-of-name-here")
      Source: (source function-name-here)
     Javadoc: (javadoc java-object-or-class-here)
    Examples from clojuredocs.org: [clojuredocs or cdoc]
              (user/clojuredocs name-here)
              (user/clojuredocs "ns-here" "name-here")
boot.user=> (+ 1 2)
IllegalStateException Attempting to call unbound fn: #'net.cgrand.parsley.fold/cat
        clojure.lang.Var$Unbound.throwArity (Var.java:43)
        net.cgrand.parsley.fold/stitch (fold.clj:124)
        net.cgrand.parsley/stepper/self--6726 (parsley.clj:37)
        reply.parsing/reparse (parsing.clj:25)
        reply.parsing/parsed-forms (parsing.clj:62)
        reply.eval-modes.nrepl/run-repl (nrepl.clj:132)
        reply.eval-modes.nrepl/main (nrepl.clj:234)
        reply.main/launch-nrepl/fn--7418 (main.clj:78)
        clojure.core/with-redefs-fn (core.clj:7142)
        reply.main/launch-nrepl (main.clj:77)
        boot.repl-client/client (repl_client.clj:11)
        clojure.lang.Var.invoke (Var.java:379)
Bye for now!
```

**Comment by micha** [comment]
This is related to cgrand/parsley#15 via trptcolin/reply#159

**Comment by micha** [comment]
The workaround for now is to start the REPL server with Clojure 1.7 and the client with 1.6:

In the first terminal
```
$ BOOT_CLOJURE_VERSION=1.7.0-alpha4 boot repl -s wait
```
and in the other terminal
```
$ boot repl -c
```

**Comment by micha** [comment]
I just verified that the issue is fixed in master in parsley, so if you want you can also build that from source and install in your local Maven repo. That will also fix the problem.

**Comment by micha** [comment]
This issue appears to be fixed now. Please reopen if it's not really fixed.

**Comment by tgetgood** [comment]
This was fixed for me with boot 2.0.0-rc9 by manually installing parsley master as mentioned above, but since I upgraded to rc12 the other day I get the exact same error again. I haven't yet tried the releases in between.

Versions in case they're useful:

```
BOOT_VERSION=2.0.0-rc12
REPL-y 0.3.5, nREPL 0.2.7
Clojure 1.7.0-alpha5
OpenJDK 64-Bit Server VM 1.8.0_31-b13
```
